BotRoyaleUnity/Assets/Scripts/PartHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PartHandler : MonoBehaviour
+{
+    //keeps track of the positions where there are alive parts.
+    public bool[,] parts = new bool[9, 9];
+
+    //array to calculate all attached parts.
+
+    public bool[,] attachedParts;
+
+    public void setParts()
+    {
+        foreach (Transform child in transform)
+        {
+            int x = (int)child.position.x;
```

Comment:
does this work if you initialize the robot parent at a position other than (0, 0, 0)? transform.position returns the object's position in space, it may work better to use transform.localPosition so that you get the position relative to the parent